OpenRA.Mods.Common/Activities/Move/AttackMoveActivity.cs

```diff
@@ -42,9 +42,9 @@ protected override void OnFirstRun(Actor self)
 			if (attackMove == null)
 				return;
 
-			if (!isAssaultMove && !string.IsNullOrEmpty(attackMove.Info.AttackMoveCondition))
+			if (!isAssaultMove)
 				token = self.GrantCondition(attackMove.Info.AttackMoveCondition);
-			else if (isAssaultMove && !string.IsNullOrEmpty(attackMove.Info.AssaultMoveCondition))
+			else if (isAssaultMove)
```

Comment:
This can now simplify to just `else`, and it might make sense to invert the condition and order of the two cases?